force_wfmanager/ui/setup/mco/mco_parameter_view.py

```diff
@@ -0,0 +1,257 @@
+from traits.api import (
+    Instance, Unicode, Bool, on_trait_change, Event, Property,
+    cached_property, HasTraits, List, Button
+)
+from traitsui.api import (
+    View, Item, HGroup, ModelView, ListEditor, VGroup, InstanceEditor
+)
+
+from force_bdss.api import BaseMCOParameter, BaseMCOModel
+
+from force_wfmanager.ui.ui_utils import get_factory_name
+from force_wfmanager.ui.setup.new_entity_creator import NewEntityCreator
+
+
+class MCOParameterModelView(ModelView):
+
+    # -------------------
+    # Required Attributes
+    # -------------------
+
+    #: MCO parameter model
+    model = Instance(BaseMCOParameter)
+
+    # ------------------
+    # Dependent Attributes
+    # ------------------
+
+    #: Defines if the MCO parameter is valid or not. Updated by
+    #: :func:`verify_tree
+    #: <force_wfmanager.ui.setup.workflow_tree.WorkflowTree.verify_tree>`
+    valid = Bool(True)
+
+    #: An error message for issues in this modelview. Updated by
+    #: :func:`verify_tree
+    #: <force_wfmanager.ui.setup.workflow_tree.WorkflowTree.verify_tree>`
+    error_message = Unicode()
+
+    #: Event to request a verification check on the workflow
+    #: Listens to: :attr:`model.name <model>` and :attr:`model.type <model>`
+    verify_workflow_event = Event
+
+    # ----------
+    # Properties
+    # ----------
+
+    #: The human readable name of the MCO parameter class
+    label = Property(Unicode(), depends_on="model.[name,type]")
+
+    # ----------
+    #    View
+    # ----------
+
+    def default_traits_view(self):
+        # Obtains traits of BaseMCOParameter that will always be present
+        base_traits = [Item('model.name'),
+                       Item('model.type')]
+
+        # Obtains user-created traits that have been tagged as visible
+        user_traits = [
```

Comment:
Typically, user made plugins define MCO parameters with various extra traits that contribute to the MCO optimisation process. If these are tagged as `visible=True` (default) then they will be displayed.